perspective.el

```diff
@@ -618,35 +593,36 @@ See also `persp-switch' and `persp-add-buffer'."
         ;;       rather than just the current buffer?
         ((eq buffer (current-buffer)) (bury-buffer))
         (t (bury-buffer buffer)))
-  (setf (persp-buffers persp-curr) (remq buffer (persp-buffers persp-curr))))
+  (setf (persp-buffers (frame-parameter nil 'persp-curr)) (remq buffer (persp-buffers (frame-parameter nil 'persp-curr)))))
 
 (defun persp-kill (name)
   "Kill the perspective given by NAME.
 
 Killing a perspective means that all buffers associated with that
 perspective and no others are killed."
   (interactive "i")
-  (if (null name) (setq name (persp-prompt (persp-name persp-curr) t)))
+  (if (null name) (setq name (persp-prompt (persp-name (frame-parameter nil 'persp-curr)) t)))
   (with-perspective name
     (run-hooks 'persp-killed-hook)
-    (mapc 'persp-remove-buffer (persp-buffers persp-curr))
-    (setf (persp-killed persp-curr) t))
-  (remhash name perspectives-hash)
+    (mapc 'persp-remove-buffer (persp-buffers (frame-parameter nil 'persp-curr)))
+    (setf (persp-killed (frame-parameter nil 'persp-curr)) t))
+  (remhash name (frame-parameter nil 'perspectives-hash))
   (persp-update-modestring)
-  (when (equal name (persp-name persp-last))
-    (setq persp-last nil))
-  (when (equal name (persp-name persp-curr))
+  (when (equal name (persp-name (frame-parameter nil 'persp-last)))
+    (set-frame-parameter nil 'persp-last nil))
+  (when (equal name (persp-name (frame-parameter nil 'persp-curr)))
+    (persp-switch (persp-find-some))
     ;; Don't let persp-last get set to the deleted persp.
-    (persp-frame-local-let ((persp-last persp-last)) (persp-switch (persp-find-some)))))
+    (set-frame-parameter nil 'persp-last (frame-parameter nil 'nil))))
```

Comment:
I'm worried that setting `persp-last` to `nil` will break existing behavior. Can you set it to `(persp-find-some)` in its frame instead, as before?

Reply:
I did some improvements around this, but I am not sure if it's the correct approach.